TLM/TLM/UI/SubTools/TimedTrafficLightsTool.cs

```diff
@@ -99,9 +100,37 @@ public TimedTrafficLightsTool(TrafficManagerTool mainTool)
         }
 
         public override void OnPrimaryClickOverlay() {
-            if (HoveredNodeId <= 0 || nodeSelectionLocked) {
+            if (HoveredNodeId <= 0 || nodeSelectionLocked || !Flags.MayHaveTrafficLight(HoveredNodeId)) {
                 return;
             }
+            bool ctrlDown = Input.GetKey(KeyCode.LeftControl) || Input.GetKey(KeyCode.RightControl);
+            if(ctrlDown) {
+                AutoTimedTrafficLights.ErrorResult res = AutoTimedTrafficLights.Setup(HoveredNodeId);
+                if (res != AutoTimedTrafficLights.ErrorResult.Success) {
+                    string message;
+                    switch (res) {
+                        case AutoTimedTrafficLights.ErrorResult.NoJunction:
+                            message = "Dialog.Text:This is not a junction";
+                            break;
+                        case AutoTimedTrafficLights.ErrorResult.NoNeed:
+                            message = "Dialog.Text:There is no need";
+                            break;
+                        case AutoTimedTrafficLights.ErrorResult.TTLExists:
+                            message = "Dialog.Text:Timed traffic lights already exists";
```

Comment:
Make key a little more specific, for example "Dialog.Error:Auto TL already exists"